subsys/bluetooth/host/hci_core.c

```diff
@@ -3798,9 +3798,9 @@ int bt_disable(void)
 
 	bt_monitor_send(BT_MONITOR_CLOSE_INDEX, NULL, 0);
 
-#if defined(CONFIG_BT_EXT_ADV) || defined(CONFIG_BT_BROADCASTER)
+#if defined(CONFIG_BT_EXT_ADV) && defined(CONFIG_BT_BROADCASTER)
 	bt_adv_reset_adv_pool();
-#endif /* CONFIG_BT_EXT_ADV || CONFIG_BT_BROADCASTER */
+#endif /* CONFIG_BT_EXT_ADV && CONFIG_BT_BROADCASTER */
```

Comment:
You sure about this? 

`bt_adv_reset_adv_pool` seems to be fine without `CONFIG_BT_EXT_ADV`

```c
void bt_adv_reset_adv_pool(void)
{
#if defined(CONFIG_BT_EXT_ADV)
	(void)memset(&adv_pool, 0, sizeof(adv_pool));
#endif /* defined(CONFIG_BT_EXT_ADV) */

	(void)memset(&bt_dev.adv, 0, sizeof(bt_dev.adv));
}
```

Reply:
No, I discussed this with Sean and will update